README.md

```diff
@@ -550,6 +550,32 @@ console.log(newName); // ['Ryan', 'McDermott'];
 ```
 **[⬆ back to top](#table-of-contents)**
 
+### Avoid Side Effects pt.2
+Side effects could also occur from inside a function. In Javascript, function arguments 
+are always passed by value except when they(functions) are passed reference values such as 
+objects and arrays. In that case, we should be carefull not to change any of these 
+argument's properties. A possible solution would be to always clone the variable,
+edit it and return the clone.
```

Comment:
We probably want to mention, as a caveat, that there are some times where people want to modify the passed in object. Also, we might want to mention the performance implications of cloning large objects. All in all, I agree with the sentiment though that intentional side effects can be good, but oftentimes it's better to avoid them altogether if you can.